### PR TITLE
Reuse outdoor device logic for setup mode interface selection, plus some cleanup

### PR DESCRIPTION
**Pull request by NeoRaider**
Opening as draft, as this is completely untested so far. I intend to try out this on a few devices tomorrow or on the weekend.

### CODE REVIEW

**Comment by NeoRaider** [comment]
Tested on UniFi Outdoor+.

File: package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua [hunk 1]
```diff
@@ -66,24 +70,36 @@ function M.device_supports_wpa3()
...
 	return supports_mfp
 end
 
+function M.device_uses_11a(uci)
+	local ret = false
+
+	uci:foreach('wireless', 'wifi-device', function(radio)
+		if radio.hwmode == '11a' or radio.hwmode == '11na' then
+			ret = true
+			return false
```
Comment (mweinelt) [inline package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua:98]: What is the significane of returning false here?

File: package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua [hunk 2]
```diff
@@ -26,7 +26,10 @@ end
 function M.is_outdoor_device()
 	if M.match('ar71xx', 'generic', {
 		'bullet-m',
+		'cpe210',
```
Comment (mweinelt) [inline package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua:29]: Does that also match the CPE220v3?

Comment (NeoRaider) [reply package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua:29]: No, I'll add it.

File: package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua (hunk 1 above)
Comment (NeoRaider) [reply package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua:98]: Stop the `foreach` early without going through the remaining entries.

**Pull request merged**